File: c/2.1.4/2.1.4_Potentiometer.c
/**********************************************************************
* Filename    : 2.1.4_Potentiometer.c
* Description : 
* Author      : Robot
* E-mail      : support@sunfounder.com
* website     : www.sunfounder.com
* Update      : Cavon    2016/07/01
**********************************************************************/

#include <stdio.h>
#include <wiringPi.h>
#include <pcf8591.h>

#define PCF       120

int main (void)
{
	printf("\n");
	printf("\n");
	printf("========================================\n");
	printf("|             Potentiometer            |\n");
	printf("|    ------------------------------    |\n");
	printf("|           SDA,SCL connect to         |\n");
	printf("|   the corresponding pin of PCF8591   |\n");
	printf("|                                      |\n");
	printf("| Make potentiometer to contral a led  |\n");
	printf("|                                      |\n");
	printf("|                            SunFounder|\n");
	printf("========================================\n");
	printf("\n");
	printf("\n");
	int value ;
	wiringPiSetup () ;
	// Setup pcf8591 on base pin 120, and address 0x48
	pcf8591Setup (PCF, 0x48) ;
	while(1) // loop forever
	{
		value = analogRead  (PCF + 0) ;
		printf("Value: %d\n", value);
		analogWrite (PCF + 0, value) ;
		delay (200) ;
	}
	return 0 ;
}

File: python/2.1.4_Potentiometer.py
#!/usr/bin/env python3

import PCF8591 as ADC
import time
import RPi.GPIO as GPIO

def print_message():
    print ("========================================")
    print ("|             Potentiometer            |")
    print ("|    ------------------------------    |")
    print ("|           SDA,SCL connect to         |")
    print ("|   the corresponding pin of PCF8591   |")
    print ("|                                      |")
    print ("| Make potentiometer to contral a led  |")
    print ("|                                      |")
    print ("|                            SunFounder|")
    print ("========================================\n")
    print ('Program is running...')
    print ('Please press Ctrl+C to end the program...')
    input ("Press Enter to begin\n")

def setup():
    ADC.setup(0x48)

def loop():
    status = 1
    while True:
        print ('Value:', ADC.read(0))
        Value = ADC.read(0)
        outvalue = map(Value,0,255,120,255)
        ADC.write(outvalue)
        time.sleep(0.2)
def destroy():
    ADC.write(0)

def map(x, in_min, in_max, out_min, out_max):
        '''To map the value from arange to another'''
        return (x - in_min) * (out_max - out_min) / (in_max - in_min) + out_min

if __name__ == '__main__':
    print_message()
    try:
        setup()
        loop()
    except KeyboardInterrupt: 
        destroy()

        
    
